integration/helpers.go

```diff
@@ -558,6 +560,86 @@ func (i *TeleInstance) Stop(removeData bool) error {
 	return i.Process.Wait()
 }
 
+type proxyServer struct {
+	sync.Mutex
+	count int
+}
+
+// ServeHTTP only accepts the CONNECT verb and will tunnel your connection to
+// the specified host. Also tracks the number of connections that it proxies for
+// debugging purposes.
+func (p *proxyServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// validate http connect parameters
+	if r.Method != http.MethodConnect {
+		http.Error(w, fmt.Sprintf("%v not supported", r.Method), http.StatusInternalServerError)
+		return
+	}
+	if r.Host == "" {
+		http.Error(w, fmt.Sprintf("host not set"), http.StatusInternalServerError)
+		return
+	}
+
+	// hijack request so we can get underlying connection
+	hj, ok := w.(http.Hijacker)
+	if !ok {
+		http.Error(w, "unable to hijack connection", http.StatusInternalServerError)
+		return
+	}
+	sconn, _, err := hj.Hijack()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// dial to host we want to proxy connection to
+	dconn, err := net.Dial("tcp", r.Host)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// write 200 OK to the source, but don't close the connection
+	resp := &http.Response{
+		Status:     "OK",
+		StatusCode: 200,
+		Proto:      "HTTP/1.1",
+		ProtoMajor: 1,
+		ProtoMinor: 0,
+	}
+	resp.Write(sconn)
+
+	// success, we're proxying data now
+	p.Lock()
+	p.count = p.count + 1
+	p.Unlock()
+
+	// copy from src to dst and dst to src
+	done := make(chan bool)
+	go func() {
+		io.Copy(sconn, dconn)
```

Comment:
overall it feels like reimplementation of vulcand/oxy proxy with middleware, may be just use it?